### PR TITLE
Add an extensive Makefile documentation header

### PR DESCRIPTION
**Pull request by wilzbach**
While the Makefile is growing over time, the documentation isn't.

This is an effort to improve the dlang.org's Makefile documentation.

### CODE REVIEW

File: posix.mak [hunk 1]
```diff
@@ -1,25 +1,129 @@
...
+#
+#    The respective local repositories are used.
+#   	This is very useful for testing local changes.
+#   	Individual targets include:
+#
+#    	dmd-prerelease
+#    	druntime-prerelease
+#    	phobos-prerelease
+#    	apidocs-prerelease		Ddox documentation
+#    	apidocs-serve			Starts a development Ddox webserver
```
Comment (CyberShadow) [inline posix.mak]: Perhaps clarify which of these are actually dependencies of `docs-prerelease`

File: posix.mak [hunk 2]
```diff
@@ -1,25 +1,129 @@
...
+#   	apidocs-latest			Ddox documentation
+#
+#  Options
+#  -------
+#
+#  Most commonly used options include:
+#
+#   	DIFFABLE=1			Removes inclusion of all dynamic content and timestamps
+#   	RELEASE=1			Release build (needs to be set for the `release` target)
+#   	LATEST=2.072.1		Specify the LATEST version manually. Avoids network traffic.
```
Comment (CyberShadow) [inline posix.mak]: LATEST no longer (thankfully) requires network traffic to get. It's specified in the VERSION file. I'm not sure if there is any reason to override it on the command line any more.

File: posix.mak [hunk 3]
```diff
@@ -1,25 +1,129 @@
...
+#  -------------
+#
+#   	html				Builds all HTML files and static content
+#   	pending_changelog	Collects and assembles the changelog for the next version
+#   						(This is based on references Bugzilla issues and files in the `/changelog` folders)
+#   	rebase				Rebase all DLang repos to upstream/master
+#   	pdf					Generates the D specification as a PDF
+#   	mobi				Generates the D specification as an ebook (Amazon mobi)
+#   	verbatim			Copies the Ddoc plaintext files to .verbatim files (i.e. doesn't run Ddoc on them)
+#   	rsync				Publishes the website (not used anymore)
```
Comment (CyberShadow) [inline posix.mak]: I think Martin/Andrei still use rsync when there's a problem with the auto-deploy.

If a target is actually no longer used, it probably ought to be removed.

File: posix.mak [hunk 4]
```diff
@@ -1,25 +1,129 @@
...
+#
+#   For more documentation on Ddox, see https://github.com/rejectedsoftware/ddox
+#   For more information and current blocking points of the Ddoc -> Ddox tranisition,
+#   see https://github.com/dlang/dlang.org/pull/1526
+#
+#	Building without network traffic
+#	--------------------------------
+#
+#	You can force this Makefile to use the generated DMD executable in `../dmd`
+#	instead of download a stable DMD compiler:
```
Comment (CyberShadow) [inline posix.mak]: "instead of downloading"?

File: posix.mak (hunk 3 above)
Comment (wilzbach) [reply posix.mak]: AFAICT they haven't done this in a long time. Your auto-deployment is pretty amazing!

**Pull request merged**